lms/djangoapps/courseware/courses.py

```diff
@@ -609,11 +609,7 @@ def get_courses(user, org=None, filter_=None):
         org=org,
         filter_=filter_,
     ).prefetch_related(
-        Prefetch(
-            'modes',
-            queryset=CourseMode.objects.exclude(mode_slug__in=CourseMode.CREDIT_MODES),
-            to_attr='selectable_modes',
-        ),
+        'modes',
```

Comment:
No, we are still excluding them, just prefetching differently. Previously, we were saving a prefetched and filtered set of modes to `selectable_modes` on the object (see the `to_attr`). Now, we are simply prefetching the modes field. It will work out to be the same (actually, better, since now selectable and normal modes will be prefetched). See:

```
        if only_selectable:
            if course is not None and hasattr(course, 'selectable_modes'):
                found_course_modes = course.selectable_modes
            elif exclude_credit:
                found_course_modes = found_course_modes.exclude(mode_slug__in=cls.CREDIT_MODES)
```

That's the code in `modes_for_course` that checks that uses `selectable_modes` and you can see it falls back to excluding credit all by itself (`exclude_credit` defaults to True).

But our code path was skipping that since we pass in `only_selectable=False` and we weren't getting the benefit of the credit-excluding prefetch. Now both code paths get the benefit.

But I'm also seeing another small refactor to be done. Now *nothing* is setting `selectable_modes` and that code should be removed. And the only place that passes in `exclude_credit=False` should really just be be passing in `only_selectable=False` so I should remove that too. Sigh. Yaks!